src/Application/Asset/CompleteUploadService.php

```
@@ -63,28 +68,55 @@ public function completeUpload(CompleteUploadCommand $command): CompleteUploadRe
+            $this->transactionManager->commit();
+        } catch (\Throwable $e) {
+            try {
+                $this->transactionManager->rollBack();
+            } catch (\Throwable $suppressed) {
+                // Suppressed intentionally: the primary failure $e is already
+                // captured and will be rethrown below. A rollback failure here
+                // is a secondary cleanup error and must not replace the root cause.
+                unset($suppressed);
+            }
+
+            throw RepositoryUnavailableException::forReason(self::REPOSITORY_FAILURE_REASON, $e);
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\App\Domain\Asset\Exception\RepositoryUnavailableException' in method 'markProcessingAndDispatch'. (undefined)

(StaticAccess)